gotest.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# The script does automatic checking on a Go package and its sub-packages, including:
+# 1. gofmt         (http://golang.org/cmd/gofmt/)
+# 2. golint        (https://github.com/golang/lint)
+# 3. go vet        (http://golang.org/cmd/vet)
+# 4. gosimple      (https://github.com/dominikh/go-simple)
+# 5. unconvert     (https://github.com/mdempsky/unconvert)
+# 6. race detector (http://blog.golang.org/race-detector)
+# 7. test coverage (http://blog.golang.org/cover)
+#
+# gometalinter (github.com/alecthomas/gometalinter) is used to run each static
+# checker.
+
+declare GREEN='\033[0;32m'
+declare NC='\033[0m'
+print () {
+    echo -e "${GREEN}$1${NC}"
+}
+
+# test_with_profile run test coverage on each subdirectories and merge the
+# coverage profile.
+test_with_coverage_profile() {
+    print "* Run tests with creating coverage profile:"
+
+    echo "mode: count" > profile.cov
+
+    # Standard go tooling behavior is to ignore dirs with leading underscores.
+    for dir in $(find . -maxdepth 10 \
+    -not -path './.git*' \
+    -not -path '*/_*' \
+    -not -path './cmd*' \
+    -not -path './release*' \
+    -not -path './vendor*' \
+    -type d)
+    do
+    if ls $dir/*.go &> /dev/null; then
+      go test -v -covermode=count -coverprofile=$dir/profile.tmp $dir
+
+      if [ -f $dir/profile.tmp ]; then
+        cat $dir/profile.tmp | tail -n +2 >> profile.cov
+        rm $dir/profile.tmp
+      fi
+    fi
+    done
+
+    if [ "$TRAVIS" == "true" ] ; then
+        print "* Send test coverage to travis server:"
+        # Make sure goveralls is installed and $GOPATH/bin is in your path.
+        if [ ! -x "$(type -p goveralls)" ]; then
+            print "** Install goveralls:"
+            go get github.com/mattn/goveralls
```

Comment:
Yeah, the original idea was to create the way for contributors check their code before sending the `PR`.